burgerNode.js

```diff
@@ -49,9 +49,78 @@ class BurgerNode {
     }
 
     appendPendingTransactions(pendingTransactions) {
-        pendingTransactions.forEach((transaction) => {
+        let allPendingTransactions = pendingTransactions;
+
+        if (pendingTransactions.length == 0 && this.chain.pendingTransactions.length == 0) {
+            /**
+             * Nothing to do here!
+             */
+            return this.chain.pendingTransactions;
+        }
+
+        if (this.chain.pendingTransactions.length > 0 && pendingTransactions.length > 0) {
+            /**
+             * Append operation between incoming pending transactions and own pending transactions
+             */
+            for (let currentPendingIndex = 0; currentPendingIndex < this.chain.pendingTransactions.length; currentPendingIndex++) {
+                const transaction = this.chain.pendingTransactions[currentPendingIndex];
+                for (let allPendingIndex = 0; allPendingIndex < allPendingTransactions.length; allPendingIndex++) {
+                    if (JSON.stringify(transaction) === JSON.stringify(allPendingTransactions[allPendingIndex])) {
+                        allPendingTransactions[allPendingIndex] = 0;
+                    }
+                }
+            }
+
+            allPendingTransactions = allPendingTransactions.filter((transactions) => {
+                return transactions !== 0;
+            });
+        }
+    
+        if (pendingTransactions.length > 0 && this.chain.pendingTransactions.length === 0) {
+            /**
+             * Sync the incoming pending transactions.
+             */
+            allPendingTransactions = pendingTransactions;
+        }
+
+        if (pendingTransactions.length === 0 && this.chain.pendingTransactions.length > 0) {
+            /**
+             * Nothing to append!
+             */
+            return this.chain.pendingTransactions;
+        }
+
+        /**
+         * Clear the current pending transactions to 
+         * prepare it for the appended one.
+         */
+        this.chain.pendingTransactions = [];
+        
+        allPendingTransactions.forEach((transaction) => {
             this.addPendingTransaction(transaction);
```

Comment:
because we still need to validate the transactions, so we are clearing the array first, then let the validations decide if this pending transaction is valid or not.

Reply:
the `this.addPendingTransaction(transaction)` encapsulates the validation function for the transaction object.